tests/dragonfly/connection_test.py

```diff
@@ -0,0 +1,176 @@
+import asyncio
+import aioredis
+import async_timeout
+#from conftest import DATABASE_INDEX
+
+
+def verify_response(monitor_response: dict, key: str, value: str) -> bool:
+    if monitor_response is None:
+        return False
+    if monitor_response["db"] == 1 and monitor_response["client_type"] == "tcp":
+        return key in monitor_response["command"] and value in monitor_response["command"]
+    else:
+        return False
+
+
+async def monitor_cmd(mon: aioredis.client.Monitor, messages: dict):
+    success = None
+    async with mon as monitor:
+        try:
+            for key, value in messages.items():
+                try:
+                    async with async_timeout.timeout(1):
+                        response = await monitor.next_command()
+                        success = verify_response(
+                            response, key, value)
+                        if not success:
+                            return False, f"failed on the verification of the message {response} at {key}: {value}"
+                        await asyncio.sleep(0.01)
+                except asyncio.TimeoutError:
+                    pass
+            return True, "monitor is successfully done"
+        except Exception as e:
+            return False, f"stopping monitor on {e}"
+
+
+async def run_monitor(messages: dict, pool: aioredis.ConnectionPool):
+    cmd1 = aioredis.Redis(connection_pool=pool)
+    conn = aioredis.Redis(connection_pool=pool)
+    monitor = conn.monitor()
+    future = asyncio.create_task(monitor_cmd(monitor, messages))
+    success = True
+    for key, val in messages.items():
+        res = await cmd1.set(key, val)
+        if not res:
+            success = False
+            break
+        await asyncio.sleep(1)
+    await asyncio.sleep(1)
```

Comment:
why do you call `sleep` here and above?

Reply:
I tried to make this work for redis as well, and while with redis this test was unstable (it was closing the connection all the time), it was working aways for dragonfly. I can try and remove these sleeps. But you need a way to let the scheduler run other tasks (its not threads), I just that on all examples in python they were using sleeps to let other task run. Maybe they have yield for this - I need to take a look, In any case we can make it a much shorter sleeps

Reply:
I just tried to understand - I do not know what's right. (When I see sleeps in tests I suspect it will be fragile).
Then just add short comments to explain the reason.

Reply:
I remove the sleep from 1 second to to millisecond, I don't see other way to make sure that there is a way to ensure "context switch" between tasks (there is nothing like yield the task in the API).